crate/server/src/database/cached_sqlite_struct.rs

```diff
@@ -273,7 +273,8 @@ impl KMSSqliteCache {
                 let freeable_cache_id = freeable_sqlites.push(id)?;
 
                 // Add it to the SqliteCache
-                let mut item = KMSSqliteCacheItem::new(pool, key.to_owned(), freeable_cache_id);
+                let key: [u8; 32] = key.as_bytes().try_into()?;
+                let mut item = KMSSqliteCacheItem::new(pool, key, freeable_cache_id);
```

Comment:
I would try to pass the `SymmetricKey` to `KMSSqliteCacheItem::new()` instead. Otherwise you create yet an other non-zeroizing copy of the key in memory.

One could argue that the KMS is intended to run in a safe place, which would be true. So I guess this is pretty much up to you :shrug:

Reply:
done here https://github.com/Cosmian/kms/pull/16/commits/7ebe6f13a6a2589f44430db0be9dd43175ac237b